### PR TITLE
Merge OpenBSD platform support into 5.3.

### PR DESCRIPTION
**Pull request by 3405691582**
Instead of maintaining these commits as platform-specific port patches until they eventually land in the release branch, merge them all into the 5.3 branch.

Most of these commits are cherry-picked verbatim from `main`. Some are mutated slightly to account for the branch differences; these commits have a `[5.3]` tag. One minor new commit is necessary for one test.

Some of these commits may understandably not be appropriate to merge into a release branch; please let me know and I'll remove them; these will be maintained as local patches.

### CODE REVIEW

**Comment by buttaface** [comment]
Good luck, there haven't been any merge windows on the 5.3 branch and [nothing has been merged to it in a month](https://github.com/apple/swift/commits/release/5.3).

File: test/lit.cfg [hunk 1]
```diff
@@ -338,6 +344,11 @@ if run_vers.endswith('-simulator'):
 else:
     run_environment=''
 
+target_arch = run_cpu
+if run_os == 'openbsd' and run_cpu == 'amd64':
+   target_arch = run_cpu
```
Comment (buttaface) [inline test/lit.cfg:349]: Redundant as already set above, unless you meant to put it after the following line.

**Review by finagolfin — APPROVED** [review]
Everything is scoped for OpenBSD, other than some test tweaks which all look fine, so should be okay to merge.

**Comment by 3405691582** [comment]
Maybe it would be better if I checked this against the 5.4 branch now instead. I'll look into it.

**Comment by CodaFi** [comment]
Can y’all explain why you need these changes in the release branch?

**Comment by 3405691582** [comment]
Otherwise, the prospective port needs to maintain the contents of this PR as individual patches atop the release branch and have to be maintained on the port side. OpenBSD ports prefer to be built atop a release branch, not master.

However, it would be _nice_ to ensure that the patchset were as small as possible, but not _necessary_, so if this is somehow not a good idea, we'll just maintain the patches manually.

I hadn't actually noticed that the 5.4 branch was cut, so maybe the situation has changed.

**Comment by 3405691582** [comment]
Closing this; 5.4 will be a much better candidate. If there are diffs required to upstream to that release branch, that should be a new PR.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
Was this merged into 5.4?

**Comment by 3405691582** [comment]
No, but Swift builds at HEAD on the platform now.

Note that just having Swift doesn't let you build and compile iOS applications.

**Pull request closed without merging**